File: src/com/juhnowski/onthology/logical/LogicalPicture.java
package com.juhnowski.onthology.logical;

import com.juhnowski.onthology.*;
import com.juhnowski.onthology.Form;

import java.util.HashMap;

public class LogicalPicture extends Model<Reality> {
    public boolean value;
    public boolean isImagined;
    public boolean isTheReal;
    public boolean isCorrect;

    public LogicalPicture(){}
    public LogicalPicture(Structure structure){
        this.structure = structure;
    }

    Situation situation;
    Structure structure;
    HashMap<Element, TheObject> elements;
    LogicForm<Element> pictorialForm;

    public HashMap<Element,Thing> pictorialRelationship;
}

File: src/com/juhnowski/onthology/logical/LogicalPicture.kt
package com.juhnowski.onthology.logical

import com.juhnowski.onthology.*

import java.util.HashMap

abstract class LogicalPicture<T> : Model<T> {
    var value: Boolean = false
    var isImagined: Boolean = false
    var isTheReal: Boolean = false
    var isCorrect: Boolean = false

    internal var situation: Situation? = null
    internal var structure: Structure<*>
    internal var elements: HashMap<Element, TheObject>? = null
    internal var pictorialForm: LogicForm<Element>? = null
    internal var sense: Sense? = null

    var pictorialRelationship: HashMap<Element, Thing>? = null


    constructor() {}
    constructor(structure: Structure<*>) {
        this.structure = structure
    }

    fun represent(): Sense? {
        return sense
    }

    abstract fun agreement(realitySense: Sense): Boolean


}
